### PR TITLE
Add a check for latitude and a fix for longitude in construtors

### PR DESCRIPTION
**Pull request by eliascarv**

### CODE REVIEW

File: src/crs/geographic.jl [hunk 1]
```diff
@@ -15,7 +15,7 @@ ndims(::Type{<:Geographic}) = 3
     GeodeticLatLon(lat, lon)
     GeodeticLatLon{Datum}(lat, lon)
 
-Geodetic latitude `lat ∈ [-90°,90°]` and longitude `lon ∈ [-180°,180°]` in angular units (default to degree)
+Geodetic latitude `lat ∈ [-90°,90°]` and longitude `lon ∈ [-180°,180°[` in angular units (default to degree)
```
Comment (juliohm) [inline src/crs/geographic.jl:18]:
```suggestion
Geodetic latitude `lat ∈ [-90°,90°]` and longitude `lon ∈ [-180°,180°)` in angular units (default to degree)
```

File: src/crs/geographic.jl [hunk 2]
```diff
@@ -82,7 +82,7 @@ const LatLon = GeodeticLatLon
     GeodeticLatLonAlt(lat, lon, alt)
     GeodeticLatLonAlt{Datum}(lat, lon, alt)
 
-Geodetic latitude `lat ∈ [-90°,90°]` and longitude `lon ∈ [-180°,180°]` in angular units (default to degree)
+Geodetic latitude `lat ∈ [-90°,90°]` and longitude `lon ∈ [-180°,180°[` in angular units (default to degree)
```
Comment (juliohm) [inline src/crs/geographic.jl:85]:
```suggestion
Geodetic latitude `lat ∈ [-90°,90°]` and longitude `lon ∈ [-180°,180°)` in angular units (default to degree)
```

File: src/crs/geographic.jl [hunk 3]
```diff
@@ -154,7 +154,7 @@ const LatLonAlt = GeodeticLatLonAlt
     GeocentricLatLon(lat, lon)
     GeocentricLatLon{Datum}(lat, lon)
 
-Geocentric latitude `lat ∈ [-90°,90°]` and longitude `lon ∈ [-180°,180°]` in angular units (default to degree)
+Geocentric latitude `lat ∈ [-90°,90°]` and longitude `lon ∈ [-180°,180°[` in angular units (default to degree)
```
Comment (juliohm) [inline src/crs/geographic.jl:157]:
```suggestion
Geocentric latitude `lat ∈ [-90°,90°]` and longitude `lon ∈ [-180°,180°)` in angular units (default to degree)
```

File: src/utils.jl [hunk 4]
```diff
@@ -37,6 +37,25 @@ function atanpos(y, x)
   ifelse(α ≥ zero(α), α, α + oftype(α, 2π))
 end
 
+"""
+    checklat(lat)
+
+Checks if the latitude is in the range `[-90°,90°]` and returns it if yes or throws an error otherwise.
```
Comment (juliohm) [inline src/utils.jl:43]:
```suggestion
Checks if the latitude is in the range `[-90°,90°]` before returning it.
```

File: src/utils.jl [hunk 5]
```diff
@@ -37,6 +37,25 @@ function atanpos(y, x)
...
+  if abs(lat) > 90u"°"
+    throw(ArgumentError("the latitude must be in the range [-90°,90°]"))
+  end
+  lat
+end
+
+"""
+    fixlon(lon)
+
+Fix the longitude to be in the range `[-180°,180°[`.
```
Comment (juliohm) [inline src/utils.jl:55]:
```suggestion
Fix the longitude to be in the range `[-180°,180°)`.
```

**Review by juliohm — APPROVED** [review]

**Comment by disberd** [comment]
@juliohm, @eliascarv I could not see any discussion here so may I ask why the check on the latitude was removed from this PR?

**Comment by juliohm** [comment]
Hi @disberd I believe that we removed the check in the lat coordinate because we were concerned it could cause performance issues. We never actually benchmarked the removal of the check. Do you need to check lat often? Can you help with investigating the time difference?

The longitude is no longer a problem since we wrap around the globe now with an arithmetic expression that doesn't introduce CPU branches.

**Pull request merged**